### PR TITLE
Clean up and standardize analysis process

### PR DESCRIPTION
**Pull request by jimmybuffi**
## Description

Since Colab has been updated to python 3.9, made some updates to typing syntax to be closer to python 3.9 standards. Also added lots of typing and preliminary doc strings.

The primary goal of this PR was to create a standard, simple, analysis flow, highlighted in the RebootMotionAnalysis notebook. the changes I made will I think break the old notebook, so once this is good to go, we should probably delete the old one and make RebootMotionAnalysis  the new standard one

While we're at it, we might want to come up with a better name for this repository and project haha

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (enhances currently existing functionality or changes it in some way that is non-breaking and also not addressing a bug)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] My code requires a new environment variable or other globally accessible value.

### CODE REVIEW

File: colab_notebook_utils/datatypes.py [hunk 1]
```diff
@@ -1,4 +1,82 @@
...
+
+        mocap_type = self.mocap_types[0]
+
+        return f"s3://reboot-motion-{self.org_id}/population/{mocap_type}/{self.movement_type}/" \
+               f"{self.file_type}/000000_{self.movement_type}_{self.handedness}_"
+
+
+@dataclass
+class PlayerMetadata:
+    org_player_ids: Optional[list[str]] = None
```
Comment (nkgupta1) [inline colab_notebook_utils/datatypes.py:55]: if you want to default this to an empty list, you can do 
`org_player_ids: List[str] = field(default_factory=list)`

File: colab_notebook_utils/utils.py [hunk 2]
```diff
@@ -28,7 +28,8 @@ def invoke_lambda(
     lambda_function_name: Functions,
     invocation_type: InvocationTypes,
     lambda_payload: str,
-) -> Dict:
+) -> dict:
```
Comment (nkgupta1) [inline colab_notebook_utils/utils.py:31]: is there a particular reason to change the type hinting?

Comment (jimmybuffi) [reply colab_notebook_utils/utils.py:31]: yeah, from what I understand, the python community is trying to move away from lots of these explicit typing classes over time, so just trying to get ahead of the curve, but not necessary if we don't want

File: colab_notebook_utils/datatypes.py (hunk 1 above)
Comment (jimmybuffi) [reply colab_notebook_utils/datatypes.py:55]: ah ok cool, for now I actually have None being used explicitly in other areas to denote nothing was entered, but in the future we could use an empty list

File: colab_notebook_utils/utils.py (hunk 2 above)
Comment (nkgupta1) [reply colab_notebook_utils/utils.py:31]: Oh yeah you are right. I didn't realize that, thanks for letting me know.

**Comment by nkgupta1** [comment]
Please squash commits

**Review by nkgupta1 — APPROVED** [review]

**Pull request merged**